packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -105,36 +121,57 @@ const loadTSConfigFile = async (
   return configObject;
 };
 
-let registeredCompilerPromise: Promise<Service>;
+let registeredCompilerPromise: Promise<TsLoader>;
```

Comment:
I wonder if this needs to be a `Map<TsLoaderModule, Promise<TsLoader>>`? I'm thinking in cases where multi projects are used, and some project uses a different TS loader

Reply:
I have added integration tests for different loaders in different projects and they seem to be working fine [d2c3a8d](https://github.com/jestjs/jest/pull/15190/commits/d2c3a8dd490537b64bfb06a1cbacff03e8c047fa)
do we still need to save the loader and doesn't allow using a different one?